src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -198,5 +206,13 @@ private static int RunInteractive(string scriptName)
             var result = command.Execute();
             return result.ExitCode;
         }
+
+        private class RunCommandFactory : ICommandFactory
```

Comment:
Should we put this in its own internal class or something like that? Just asking, don't have a strong opinion on it.